src/accelerate/accelerator.py

```diff
@@ -488,7 +498,11 @@ def __init__(
             )
 
         # TODO: S1ro - this is probably gonna be a problem with other fp8 backends too
-        if self.fp8_backend == "AO" and self.state.fsdp_plugin.cpu_ram_efficient_loading:
+        if (
+            self.fp8_backend == FP8BackendType.AO
+            and self.state.distributed_type == DistributedType.FSDP
+            and self.state.fsdp_plugin.cpu_ram_efficient_loading
+        ):
```

Comment:
I was about to submit a PR that made this exact change :)

Reply:
Related: https://github.com/huggingface/transformers/pull/39370

Reply:
yeah IIRC this was where some of the existing fp8 tests were failing